docs/section5/Rule5-8.md

```diff
@@ -0,0 +1,45 @@
+
+# Envelope - Rule 5-8  
+
+**Rule ID:** 5-8  
+**Rule Description:**  Fenestration U-factors for new buildings, existing buildings, and additions shall match the appropriate requirements in Tables G3.4-1 through G3.4-8 for the applicable glazing percentage for Uall.  
+**Rule Assertion:** Baseline RMR = expected value  
+**Appendix G Section:** Section G3.1-5(d) Building Envelope Modeling Requirements for the Baseline building  
+**Appendix G Section Reference:** Tables G3.4-1 to G3.4-8  
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:**  None  
+
+**Manual Check:** None  
+**Evaluation Context:** Each Data Element  
+**Data Lookup:** Tables G3.4-1 to G3.4-8  
+
+## Rule Logic:  
+
+- Get building climate zone: ```climate_zone = B_RMR.weather.climate_zone```  
+
+- For each building segment in the Baseline model: ```for building_segment_baseline in B_RMR.building.building_segments:```  
+
+  - For each thermal_block in building segment: ```for thermal_block_baseline in building_segment_baseline.thermal_blocks:```  
+
+  - For each zone in thermal block: ```for zone_baseline in thermal_block_baseline.zones:```  
+
+  - For each space in thermal zone: ```for space_baseline in zone_baseline.spaces:```  
+
+    - Get surfaces in space: ```for surface_baseline in space_baseline.surfaces:```  
+
+      - Check if surface is exterior wall: ```if ( surface_baseline.classification == "WALL" ) AND ( surface_baseline.adjacent_to == "AMBIENT" ): wall_surface_baseline = surface_baseline```  
+
+        - Calculate the total wall area: ```total_wall_area_baseline += wall_surface_baseline.area```  
+
+        - Get the U-factor for vertical glazings: ```window_performance_value_baseline.append(window.u_factor for window in wall_surface_baseline.fenestration_subsurfaces)```  
+
+        - Calculate the total vertical glazing area: ```total_window_area_baseline += sum(window.area for window in wall_surface_baseline.fenestration_subsurfaces)```  
+
+        - Calculate the window to wall ratio: ```window_wall_ratio_baseline = total_window_area_baseline / total_wall_area_baseline```  (Note XC, can we get the WWR and skylight roof ratio from Rule 5-7 instead of calculating it here?)
```

Comment:
@xingcx I don't think we have anyway of saying that other than as an applicability check but I think it could be computed in another rule and referenced here.

Reply:
@JasonGlazer i never realized that the % wall in the baseline is determined for each surface and not for a building segment. Good to know

Reply:
Can we make calculate_wwr and calculate_srr as function, similar to match_data_element()?

Reply:
It is supposed to be for the total building (which was my interpretation of the pseudocode but I do see some ambiguity). The symbol += means that it is accumulating it through the loops. Maybe we should have two sets of loops, one that adds it up and then the fraction is calculated and then another loop when it is applied.

Reply:
@xingcx Your idea of a function to do this is a good idea. It seems to be needed for a bunch of rules so maybe a single function that is defined solves the problem.

Reply:
A single function and two loops would make this much clearer. Thank you!